app/controllers/api/v1/service_offerings_controller.rb

```diff
@@ -19,34 +19,34 @@ def applied_inventories
       def send_message_and_generate_task_for(operation_type)
         operation_params = send("params_for_#{operation_type}".to_sym)
 
-        logger.info("Entering method send_message_and_generate_task_for operation #{operation_type} with params #{operation_params.inspect}")
+        logger.info("ServiceOffering##{operation_type}: Entering method send_message_and_generate_task_for with params #{operation_params.inspect}")
 
         service_offering = model.find(operation_params[:service_offering_id].to_i)
 
         source_type = retrieve_source_type(service_offering)
-        logger.info("Retrieved SourceType(id: #{source_type.id}, name: #{source_type.name}), ServiceOffice(id: #{service_offering.source.id})")
+        logger.info("ServiceOffering##{operation_type}: Retrieved SourceType(id: #{source_type.id}, name: #{source_type.name}), ServiceOffering(id: #{service_offering.id}, source_ref: #{service_offering.source_ref})")
 
         task = Task.create!(:tenant => service_offering.tenant, :state => "pending", :status => "ok")
-        logger.info("Task(id: #{task.id}) created.")
+        logger.info("ServiceOffering##{operation_type}: ServiceOffering(id: #{service_offering.id}, source_ref: #{service_offering.source_ref}): Task(id: #{task.id}) created.")
 
         payload = send("payload_for_#{operation_type}".to_sym, task, service_offering)
 
-        logger.info("Task(id: #{task.id}): Publishing event(ServiceOffering.#{operation_type}) to kafka")
+        logger.info("ServiceOffering##{operation_type}: Task(id: #{task.id}), ServiceOffering(id: #{service_offering.id}, source_ref: #{service_offering.source_ref}): Publishing event(ServiceOffering.#{operation_type}) to kafka")
 
         messaging_client.publish_topic(
           :service => "platform.topological-inventory.operations-#{source_type.name}",
           :event   => "ServiceOffering.#{operation_type}",
           :payload => payload
         )
 
-        logger.info("Task(id: #{task.id}): event(ServiceOffering.#{operation_type}) published to kafka.")
+        logger.info("ServiceOffering##{operation_type}: ServiceOffering(id: #{service_offering.id}, source_ref: #{service_offering.source_ref}), Task(id: #{task.id}): event(ServiceOffering.#{operation_type}) published to kafka.")
 
         render :json => {:task_id => task.id}
       rescue ActiveRecord::RecordNotFound => err
-        logger.error("#{err.message}")
+        logger.error("ServiceOffering##{operation_type}: #{err.message}")
         head :bad_request
       rescue StandardError => err
-        logger.error("#{task ? "Task(id: #{task.id}):" : ""} #{err.message} (#{err.class})")
+        logger.error("ServiceOffering##{operation_type}: ServiceOffering(id: #{operation_params[:service_offering_id].to_i }, source_ref: #{service_offering&.source_ref}), #{task ? "Task(id: #{task.id}):" : ""}#{err.message} (#{err.class})")
```

Comment:
task could be nil there

Reply:
I think I count with it in expression `#{task ? "Task(id: #{task.id}):" : ""}` or did I miss other place ?

Reply:
ah, I lost in in line-break, sorry :)